Plugins/BrickGrid/Source/BrickGrid/Private/BrickRenderComponent.cpp

```diff
@@ -396,24 +420,27 @@ FPrimitiveSceneProxy* UBrickRenderComponent::CreateSceneProxy()
 					for(int32 LocalVertexZ = 0; LocalVertexZ < LocalVertexDim.Z; ++LocalVertexZ)
 					{
 						const FInt3 LocalVertexCoordinates(LocalVertexX,LocalVertexY,LocalVertexZ);
+						uint32 HasAdjacentBrickOfClass[(int32)EBrickClass::Count] = { 0 };
 
-						bool HasEmptyAdjacentBrick = false;
-						bool HasNonEmptyAdjacentBrick = false;
 						for(uint32 AdjacentBrickIndex = 0;AdjacentBrickIndex < 8;++AdjacentBrickIndex)
 						{
 							const FInt3 LocalBrickCoordinates = LocalVertexCoordinates + GetCornerVertexOffset(AdjacentBrickIndex) + LocalBrickExpansion - FInt3::Scalar(1);
 							const uint32 LocalBrickIndex = (LocalBrickCoordinates.Y * LocalBricksDim.X + LocalBrickCoordinates.X) * LocalBricksDim.Z + LocalBrickCoordinates.Z;
-							if(LocalBrickMaterials[LocalBrickIndex] == EmptyMaterialIndex)
-							{
-								HasEmptyAdjacentBrick = true;
-							}
-							else
-							{
-								HasNonEmptyAdjacentBrick = true;
-							}
+							
+							if (LocalBrickMaterials[LocalBrickIndex] == EmptyMaterialIndex)
+								HasAdjacentBrickOfClass[(int32)EBrickClass::Empty] = 1;
+							else if(BrickClassByMaterial[LocalBrickMaterials[LocalBrickIndex]] == EBrickClass::Translucent)
+								HasAdjacentBrickOfClass[(int32)EBrickClass::Translucent] = 1;
+							else if (BrickClassByMaterial[LocalBrickMaterials[LocalBrickIndex]] == EBrickClass::Opaque)
+								HasAdjacentBrickOfClass[(int32)EBrickClass::Opaque] = 1;
```

Comment:
This can just be:
```
const uint32 BrickClass = (uint32)BrickClassByMaterial[LocalBrickMaterials[LocalBrickIndex]];
HasAdjacentBrickOfClass[BrickClass] = 1;
```